File: backend/app/routers/specialties.py
from fastapi import APIRouter, Depends
from supabase import Client

from app.core.auth import CurrentStaff, require_permission
from app.core.database import get_supabase
from app.models.schemas import Specialty, SpecialtyCreate

router = APIRouter(prefix="/specialties", tags=["specialties"])


@router.get("", response_model=list[Specialty])
def list_specialties(
    _current: CurrentStaff = Depends(require_permission("service.view")),
    db: Client = Depends(get_supabase),
):
    return db.table("specialties").select("*").eq("is_active", True).order("name_ar").execute().data


@router.post("", response_model=Specialty)
def create_specialty(
    payload: SpecialtyCreate,
    _current: CurrentStaff = Depends(require_permission("service.create")),
    db: Client = Depends(get_supabase),
):
    return db.table("specialties").insert(payload.model_dump(mode="json")).execute().data[0]

File: frontend/src/api/specialties.ts
import { api } from "./client";

export type Specialty = {
  id: string;
  department_id: string | null;
  name_ar: string;
  name_en: string;
  is_active: boolean;
};

export type SpecialtyCreate = {
  department_id?: string;
  name_ar: string;
  name_en: string;
};

export const listSpecialties = () => api.get<Specialty[]>("/specialties").then((res) => res.data);

export const createSpecialty = (payload: SpecialtyCreate) =>
  api.post<Specialty>("/specialties", payload).then((res) => res.data);
